pkg/asset/installconfig/gcp/dns.go

```diff
@@ -34,41 +40,93 @@ func GetPublicZone(ctx context.Context, project, baseDomain string) (*dns.Manage
 	return dnsZone, nil
 }
 
+// GetPublicZoneByID returns a DNS managed zone from the provided project which matches the ID.
+func GetPublicZoneByID(ctx context.Context, project, id string) (*dns.ManagedZone, error) {
+	client, err := NewClient(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	dnsZone, err := client.GetPublicDNSZoneByID(ctx, project, id)
+	if err != nil {
+		return nil, err
+	}
+	return dnsZone, nil
+}
+
 // GetBaseDomain returns a base domain chosen from among the project's public DNS zones.
-func GetBaseDomain(project string) (string, error) {
+func GetBaseDomain(project string) (*Zone, error) {
 	client, err := NewClient(context.TODO())
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
 	defer cancel()
 
-	publicZones, err := client.GetPublicDomains(ctx, project)
+	publicZonesMap, err := client.GetPublicDomains(ctx, project)
```

Comment:
nit: the variable name doesn't need to specify the type.

Reply:
I have changed the variable name.

Reply:
it still says `publicZonesMap` ??